addon/components/plaid-bar/component.js

```diff
@@ -0,0 +1,75 @@
+import Ember from 'ember';
+import GroupElement from '../../mixins/group-element';
+import { select } from 'd3-selection';
+
+const {
+  Component,
+  getProperties,
+  run: { scheduleOnce }
+} = Ember;
+
+const PlaidBarComponent = Component.extend(GroupElement, {
+  layout: null,
+
+  /**
+    xScale function
+
+    @public
+    @type {D3 Scale}
+  */
+  xScale: null,
+
+  /**
+    yScale function
+
+    @public
+    @type {D3 Scale}
+  */
+  yScale: null,
+
+  /**
+    Values to render bars from. These should be the same as those used
+    for the domains of the scaling functions.
+
+    @public
+    @type {Array}
+  */
+  values: [],
+
+  fill: 'black',
+
+  fillOpacity: 1.0,
+
+  height: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.groupElement = select(this.element);
+    scheduleOnce('afterRender', this, this.drawBars);
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    scheduleOnce('afterRender', this, this.drawBars);
+  },
+
+  drawBars() {
+    let { values, xScale, yScale, height, fill, fillOpacity } =
+      getProperties(this, 'values', 'xScale', 'yScale', 'height', 'fill', 'fillOpacity');
+
+    this.groupElement.selectAll('.bar').data(values).enter().append('rect')
+      .attr('class', 'bar')
+      .attr('x', (d) => xScale(d[0]))
+      .attr('width', () => xScale.bandwidth())
+      .attr('y', (d) => yScale(d[1]))
+      .attr('height', (d) => height - yScale(d[1]))
```

Comment:
You could actually get away without it needing to know the height by using `yScale.range()[1]`, which has the added benefit that you can reverse the bars by reversing the range (same as the line chart)